### PR TITLE
Add six to the mix and fix raw_input() and xrange() for Python 3

### PR DESCRIPTION
**Pull request by cclauss**
__Ready for review.__

https://pythonhosted.org/six will be our Python 2 / 3 compatibility module.

Also removed backslashs (\\) to avoid the problem when a whitespace character is placed to the right of the backslash:  Your script breaks but the reader can not see why it is broken.
* https://www.python.org/dev/peps/pep-0008/#id19
* https://www.python.org/dev/peps/pep-0008/#id28

On raw_input() --> input():
* Use `input().strip()` to remove leading or trailing whitespace.
* Use `input.strip().lower()` when looking for specific text to be kind to users with capslock on

### CODE REVIEW

**Comment by cclauss** [comment]
Now you see why I wanted the PR checked in.  These issues are trickeyer...  But you are getting them.  Keep going.

File: artemis/examples/drunkards_walk_example.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def demo_drunkards_walk(n_steps=500, n_drunkards=5, homing_instinct = 0, n_dim=2
     for t in xrange(1, n_steps+1):
         drunkards[t] = drunkards[t-1]*(1-homing_instinct) + rng.randn(n_drunkards, n_dim)
         if t%100==0:
-            print 'Status at step {}: Mean: {}, STD: {}'.format(t, drunkards[t].mean(), drunkards[t].std())
+            print ('Status at step {}: Mean: {}, STD: {}'.format(t, drunkards[t].mean(), drunkards[t].std()))
     return drunkards
```
Comment (cclauss) [inline artemis/examples/drunkards_walk_example.py]: `print (` --> `print(`  # lose the space

**Comment by petered** [comment]
jaysus

**Comment by petered** [comment]
thaaar we go

**Pull request merged**